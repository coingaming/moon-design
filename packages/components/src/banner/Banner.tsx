import * as React from 'react';
import { IconError, IconClose, IconWarning } from '@heathmont/sportsbet-assets';
import { rem, mq } from '@heathmont/sportsbet-utils';
import styled from 'styled-components';

type BannerProps = {
  message: any;
  isOpen?: boolean;
  action?: any;
  autoHideDuration?: number; // ms
  onDismiss?: any;
  status: 'error' | 'warning';
  position: 'bottom';
  isCloseable?: boolean;
};

const BannerWrapper = styled.div<any>(
  ({
    theme: { color, space, zIndex, breakpoint },
    isCloseable,
    position = true,
  }) => [
    position === 'bottom' && {
      position: 'fixed',
      left: 0,
      bottom: 0,
      zIndex: zIndex.dialog,
      margin: space.default,
    },
    {
      minWidth: rem(280),
      width: `calc(100% - ${space.default * 2})`,
      [mq(breakpoint.medium)]: {
        width: 'fit-content',
      },
      padding: `${space.default}px ${rem(12)}`,
      backgroundColor: color.hit[100],
      borderRadius: rem(12),
      display: 'flex',
      flexDirection: 'row',
      alignItems: 'center',
      'P,  SPAN': {
        fontSize: rem(14),
        lineHeight: rem(20),
      },
    },
    isCloseable
      ? {
          justifyContent: 'space-between',
          '& > * + *': {
            marginLeft: space.default,
          },
        }
      : { justifyContent: 'center' },
  ]
);

const IconWrapper = styled.div({
  marginRight: rem(8),
});

const IconCloseWrapper = styled.div(({ theme }) => ({
  backgroundColor: theme.color.goku[100],
  borderRadius: theme.radius.largest,
  marginLeft: rem(20),
  cursor: 'pointer',
  width: rem(24),
  height: rem(24),
  display: 'flex',
  flexDirection: 'column',
  justifyContent: 'center',
  alignItems: 'center',
}));

const Banner: React.FC<BannerProps> = ({
  message,
  onDismiss,
  isOpen = true,
  status,
  isCloseable = true,
  position,
}) => {
  const [visible, setVisible] = React.useState(isOpen);

  if (!visible) {
    return null;
  }

  return (
    <BannerWrapper position={position}>
      {status === 'error' && (
        <IconWrapper>
          <IconError fontSize="1.5rem" />
        </IconWrapper>
      )}
      {status === 'warning' && (
        <IconWrapper>
          <IconWarning fontSize="1.5rem" color="krillin.100" />
        </IconWrapper>
      )}
      {message && message}
      {isCloseable && (
        <IconCloseWrapper
          onClick={() => {
            if (onDismiss && typeof onDismiss === 'function') {
              onDismiss();
            }
            setVisible(false);
          }}
        >
          <IconClose fontSize="1rem" color="trunks.100" />
        </IconCloseWrapper>
      )}
    </BannerWrapper>
  );
};

export { BannerProps };

export default Banner;
