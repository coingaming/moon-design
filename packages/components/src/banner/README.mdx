---
name: Banner
menu: Components
route: /components/banner
---

import {
  Banner,
  BannerToggle,
  Button,
  Heading,
} from '@heathmont/sportsbet-components';
import { PropsTable } from 'docs';

# Banner

Banners are generic containers for displaying important information to the user. They usually contain a heading, descriptive text and a call to action.

```jsx react-live
<Banner minWidth="600px">
  Something
  <Heading size="foxtrot" element="h2">
    Need help?
  </Heading>
  <p>
    If you require any assistance our 24/7 live chat support is here to help
  </p>
  <Button fullWidth modifier="secondary">
    Contact Us
  </Button>
</Banner>
```

## Horizontal

Banner content can also be switched to a `horizontal` orientation.

```jsx react-live
<Banner minWidth="600px" horizontal>
  <p>Don't have any bitcoins?</p>
  <Button fullWidth modifier="secondary">
    Buy bitcoins
  </Button>
</Banner>
```

## Closeable

`BannerToggle` (an extension of `Toggle`) can be used to offer users with the option to close a Banner.

```jsx react-live
<Banner minWidth="600px" horizontal>
  <p>Don't have any bitcoins?</p>
  <Button fullWidth modifier="secondary">
    Buy bitcoins
  </Button>
  <BannerToggle
    onClick={() => {
      console.log('Closed');
    }}
  />
</Banner>
```

```jsx react-live
<Banner minWidth="600px">
  <Heading size="foxtrot" element="h2">
    Need help?
  </Heading>
  <p>
    If you require any assistance our 24/7 live chat support is here to help
  </p>
  <Button fullWidth modifier="secondary">
    Contact Us
  </Button>
  <BannerToggle
    onClick={() => {
      console.log('Closed');
    }}
  />
</Banner>
```
