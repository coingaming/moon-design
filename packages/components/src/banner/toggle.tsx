/** @jsx jsx */
import * as React from 'react';
import { jsx, css } from '@emotion/core';
import { rem } from 'polished';
import { colors, breakpoints } from '@heathmont/sportsbet-tokens';
import { spacing, mq } from '@heathmont/sportsbet-utils';
import { Toggle } from '../';
jsx;

const bannerToggle = css({
  lineHeight: rem(16),
  fontSize: rem(30),
  color: colors.neutral[20],
  [mq(breakpoints.small)]: {
    top: spacing('medium'),
    right: spacing('medium'),
  },
});

const BannerToggle: React.FC<
  React.ButtonHTMLAttributes<HTMLButtonElement>
> = props => {
  return (
    <Toggle css={bannerToggle} {...props}>
      Ã—
    </Toggle>
  );
};

export { BannerToggle };
