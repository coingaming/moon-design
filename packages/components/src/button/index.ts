export { Button, ButtonProps } from './button';
