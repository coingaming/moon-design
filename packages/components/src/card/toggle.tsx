/** @jsx jsx */
import * as React from "react";
import { css, jsx } from "@emotion/core";
import { spacing } from "@heathmont/sportsbet-utils";
jsx;

export type CardToggleProps = {
  back?: boolean;
};

/**
 * Styles
 */
const cardToggle = css({
  position: "absolute",
  top: spacing(),
  right: spacing(),
  background: "none",
  padding: 0,
  cursor: "pointer",
  border: "none",
  /* Temporary styles until Icons implemented */
  fontSize: "0.9rem",
  lineHeight: "1.4rem"
});

/**
 * Component
 */
export const CardToggle: React.SFC<CardToggleProps> = ({ back }) => {
  const icon = !back ? "⚙️" : "↩️";

  return <button css={cardToggle}>{icon}</button>;
};

CardToggle.defaultProps = {
  back: false
};
