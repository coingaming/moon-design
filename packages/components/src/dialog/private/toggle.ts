import styled from 'styled-components';
import hideVisually from 'polished/lib/mixins/hideVisually';
import { focus, rem } from '@heathmont/moon-utils';

export const DialogToggle = styled.button(
  ({ theme: { border, color, opacity, radius, space } }) => ({
    position: 'absolute',
    top: rem(space.default),
    right: rem(space.default),
    padding: rem(space.small),
    verticalAlign: 'middle',
    fontSize: rem(space.small),
    lineHeight: 0,
    textDecoration: 'none',
    cursor: 'pointer',
    color: color.trunks[100],
    borderRadius: rem(radius.largest),
    border,
    borderColor: 'transparent',
    backgroundColor: color.gohan[100],
    ...focus(color.piccolo[100]),
    '&:disabled, &[disabled]': {
      cursor: 'not-allowed',
      opacity: opacity.disabled,
    },
  })
);

export const DialogToggleText = styled.span(hideVisually);
