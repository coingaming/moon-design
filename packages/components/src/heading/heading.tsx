/** @jsx jsx */
import * as React from 'react';
import { jsx } from '@emotion/core';
import { headingSizes, HeadingSizes } from './sizes';
jsx;

type HeadingProps = {
  size: HeadingSizes;
  element?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p';
  color?: string;
};

const Heading: React.SFC<HeadingProps> = ({
  element = 'p',
  size,
  color,
  ...rest
}) => {
  return jsx(element, {
    css: [{ color }, size && headingSizes[size]],
    ...rest,
  });
};

Heading.defaultProps = {
  color: 'inherit',
};

export { Heading, HeadingProps };
