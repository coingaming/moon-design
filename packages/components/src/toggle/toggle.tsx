import styled, { CSSObject } from '@emotion/styled';
import { spacing } from '@heathmont/sportsbet-utils';

type ToggleProps = {
  topRight?: boolean;
};

const topRightStyle: CSSObject = {
  position: 'absolute',
  top: spacing(),
  right: spacing(),
};

const Toggle = styled.button<ToggleProps>(({ topRight = true }) => {
  return {
    background: 'none',
    padding: 0,
    cursor: 'pointer',
    border: 'none',
    fontSize: '1rem',
    ...(topRight ? topRightStyle : {}),
  };
});

export { Toggle };
