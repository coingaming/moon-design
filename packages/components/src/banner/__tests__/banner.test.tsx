/* eslint-disable no-console */
import * as React from 'react';
import { create } from 'react-test-renderer';

import { Banner } from '../banner';
import { BannerToggle } from '../toggle';
import { Button } from '../../button/button';
import { Heading } from '../../heading/heading';

describe('Banner', () => {
  test('renders vertically by default', () => {
    const banner = create(
      <Banner>
        <Heading size="bravo" as="h2">
          Need help?
        </Heading>
        <p>
          If you require any assistance our 24/7 live chat support is here to
          help
        </p>
        <Button fullWidth modifier="secondary">
          Contact Us
        </Button>
      </Banner>
    );

    expect(banner).toMatchSnapshot();
  });

  test('renders horizontally', () => {
    const banner = create(
      <Banner horizontal>
        <p>Don't have any bitcoins?</p>
        <Button fullWidth modifier="secondary">
          Buy bitcoins
        </Button>
      </Banner>
    );

    expect(banner).toMatchSnapshot();
  });

  test('renders correctly with maxWidth prop', () => {
    const banner = create(
      <Banner maxWidth="600px">
        <Heading size="bravo" as="h2">
          Need help?
        </Heading>
        <p>
          If you require any assistance our 24/7 live chat support is here to
          help
        </p>
        <Button fullWidth modifier="secondary">
          Contact Us
        </Button>
      </Banner>
    );

    expect(banner).toMatchSnapshot();
  });

  describe('Closable', () => {
    test('renders with default vertical layout', () => {
      const banner = create(
        <Banner horizontal>
          <p>Don't have any bitcoins?</p>
          <Button fullWidth modifier="secondary">
            Buy bitcoins
          </Button>
          <BannerToggle
            onClick={() => {
              console.log('Closed');
            }}
          />
        </Banner>
      );

      expect(banner).toMatchSnapshot();
    });

    test('renders with horizontal layout', () => {
      const banner = create(
        <Banner>
          <Heading size="bravo" as="h2">
            Need help?
          </Heading>
          <p>
            If you require any assistance our 24/7 live chat support is here to
            help
          </p>
          <Button fullWidth modifier="secondary">
            Contact Us
          </Button>
          <BannerToggle
            onClick={() => {
              console.log('Closed');
            }}
          />
        </Banner>
      );

      expect(banner).toMatchSnapshot();
    });
  });
});
