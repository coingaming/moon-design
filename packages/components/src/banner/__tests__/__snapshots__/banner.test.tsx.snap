// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner Closable renders with default vertical layout 1`] = `
.emotion-3 {
  vertical-align: middle;
}

.emotion-1 {
  padding: 0.5rem 1.5rem;
  display: inline-block;
  vertical-align: middle;
  font-family: inherit;
  font-size: 0.875rem;
  font-weight: 600;
  line-height: 1.125rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  cursor: pointer;
  border-style: solid;
  border-width: 1px;
  border-color: transparent;
  border-radius: 4px;
  color: #fff;
  background-color: transparent;
  border-color: #0cd664;
  width: 100%;
}

.emotion-1:disabled,
.emotion-1[disabled] {
  cursor: not-allowed;
  background-color: #414f57;
  color: #677278;
}

.emotion-1.is-active:not([disabled]),
.emotion-1.has-focus:not([disabled]),
.emotion-1:hover:not([disabled]),
.emotion-1:active:not([disabled]),
.emotion-1:focus:not([disabled]) {
  color: #00e865;
  background-color: transparent;
  border-color: #00e865;
}

.emotion-1.has-focus,
.emotion-1:focus {
  outline: none;
  box-shadow: 0 0 16px #0cd664;
}

.emotion-1:disabled,
.emotion-1[disabled] {
  cursor: not-allowed;
  background-color: transparent;
  border-color: #414f57;
  color: #414f57;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-width: 100%;
  height: auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: inherit;
  background-color: #1a3442;
  border-radius: 4px;
}

.emotion-0 {
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  margin: 0;
  padding: 1rem 1.5rem;
}

.emotion-4 {
  position: absolute;
  top: 1rem;
  right: 1rem;
  background: none;
  padding: 0;
  cursor: pointer;
  border: none;
  font-size: 1rem;
  color: #8697a2;
}

@media (min-width:31.25em) {
  .emotion-4 {
    top: 1.5rem;
    right: 1.5rem;
  }
}

<div
  className="emotion-6"
>
  <div
    className="emotion-0"
  >
    <p>
      Don't have any bitcoins?
    </p>
  </div>
  <div
    className="emotion-0"
  >
    <button
      className="emotion-1"
    >
      Buy bitcoins
    </button>
  </div>
  <div
    className="emotion-0"
  >
    <button
      className="emotion-4"
      onClick={[Function]}
    >
      <svg
        className="emotion-3"
        height="1em"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.609 10L20 17.391 17.391 20 10 12.609 2.609 20 0 17.391 7.391 10 0 2.609 2.609 0 10 7.391 17.391 0 20 2.609z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`Banner Closable renders with horizontal layout 1`] = `
.emotion-5 {
  vertical-align: middle;
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-width: 100%;
  height: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: normal;
  -ms-flex-line-pack: normal;
  align-content: normal;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  font-family: inherit;
  background-color: #1a3442;
  border-radius: 8px;
  padding: 2.5rem;
}

.emotion-1 {
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  width: 100%;
}

.emotion-1:not(:last-child) {
  margin-bottom: 1rem;
}

.emotion-0 {
  color: inherit;
  font-weight: 700;
  font-size: 2.0625rem;
  line-height: 2.5625rem;
}

.emotion-3 {
  padding: 0.5rem 1.5rem;
  display: inline-block;
  vertical-align: middle;
  font-family: inherit;
  font-size: 0.875rem;
  font-weight: 600;
  line-height: 1.125rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  cursor: pointer;
  border-style: solid;
  border-width: 1px;
  border-color: transparent;
  border-radius: 4px;
  color: #fff;
  background-color: transparent;
  border-color: #0cd664;
  width: 100%;
}

.emotion-3:disabled,
.emotion-3[disabled] {
  cursor: not-allowed;
  background-color: #414f57;
  color: #677278;
}

.emotion-3.is-active:not([disabled]),
.emotion-3.has-focus:not([disabled]),
.emotion-3:hover:not([disabled]),
.emotion-3:active:not([disabled]),
.emotion-3:focus:not([disabled]) {
  color: #00e865;
  background-color: transparent;
  border-color: #00e865;
}

.emotion-3.has-focus,
.emotion-3:focus {
  outline: none;
  box-shadow: 0 0 16px #0cd664;
}

.emotion-3:disabled,
.emotion-3[disabled] {
  cursor: not-allowed;
  background-color: transparent;
  border-color: #414f57;
  color: #414f57;
}

.emotion-6 {
  position: absolute;
  top: 1rem;
  right: 1rem;
  background: none;
  padding: 0;
  cursor: pointer;
  border: none;
  font-size: 1rem;
  color: #8697a2;
}

@media (min-width:31.25em) {
  .emotion-6 {
    top: 1.5rem;
    right: 1.5rem;
  }
}

<div
  className="emotion-8"
>
  <div
    className="emotion-1"
  >
    <h2
      className="emotion-0"
    >
      Need help?
    </h2>
  </div>
  <div
    className="emotion-1"
  >
    <p>
      If you require any assistance our 24/7 live chat support is here to help
    </p>
  </div>
  <div
    className="emotion-1"
  >
    <button
      className="emotion-3"
    >
      Contact Us
    </button>
  </div>
  <div
    className="emotion-1"
  >
    <button
      className="emotion-6"
      onClick={[Function]}
    >
      <svg
        className="emotion-5"
        height="1em"
        viewBox="0 0 20 20"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12.609 10L20 17.391 17.391 20 10 12.609 2.609 20 0 17.391 7.391 10 0 2.609 2.609 0 10 7.391 17.391 0 20 2.609z"
          fill="currentColor"
          fillRule="evenodd"
        />
      </svg>
    </button>
  </div>
</div>
`;

exports[`Banner renders horizontally 1`] = `
.emotion-1 {
  padding: 0.5rem 1.5rem;
  display: inline-block;
  vertical-align: middle;
  font-family: inherit;
  font-size: 0.875rem;
  font-weight: 600;
  line-height: 1.125rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  cursor: pointer;
  border-style: solid;
  border-width: 1px;
  border-color: transparent;
  border-radius: 4px;
  color: #fff;
  background-color: transparent;
  border-color: #0cd664;
  width: 100%;
}

.emotion-1:disabled,
.emotion-1[disabled] {
  cursor: not-allowed;
  background-color: #414f57;
  color: #677278;
}

.emotion-1.is-active:not([disabled]),
.emotion-1.has-focus:not([disabled]),
.emotion-1:hover:not([disabled]),
.emotion-1:active:not([disabled]),
.emotion-1:focus:not([disabled]) {
  color: #00e865;
  background-color: transparent;
  border-color: #00e865;
}

.emotion-1.has-focus,
.emotion-1:focus {
  outline: none;
  box-shadow: 0 0 16px #0cd664;
}

.emotion-1:disabled,
.emotion-1[disabled] {
  cursor: not-allowed;
  background-color: transparent;
  border-color: #414f57;
  color: #414f57;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-width: 100%;
  height: auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-family: inherit;
  background-color: #1a3442;
  border-radius: 4px;
}

.emotion-0 {
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  margin: 0;
  padding: 1rem 1.5rem;
}

<div
  className="emotion-3"
>
  <div
    className="emotion-0"
  >
    <p>
      Don't have any bitcoins?
    </p>
  </div>
  <div
    className="emotion-0"
  >
    <button
      className="emotion-1"
    >
      Buy bitcoins
    </button>
  </div>
</div>
`;

exports[`Banner renders vertically by default 1`] = `
.emotion-5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  width: -webkit-fit-content;
  width: -moz-fit-content;
  width: fit-content;
  min-width: 100%;
  height: auto;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: normal;
  -ms-flex-line-pack: normal;
  align-content: normal;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  font-family: inherit;
  background-color: #1a3442;
  border-radius: 8px;
  padding: 2.5rem;
}

.emotion-1 {
  -webkit-box-flex: 2;
  -webkit-flex-grow: 2;
  -ms-flex-positive: 2;
  flex-grow: 2;
  width: 100%;
}

.emotion-1:not(:last-child) {
  margin-bottom: 1rem;
}

.emotion-0 {
  color: inherit;
  font-weight: 700;
  font-size: 2.0625rem;
  line-height: 2.5625rem;
}

.emotion-3 {
  padding: 0.5rem 1.5rem;
  display: inline-block;
  vertical-align: middle;
  font-family: inherit;
  font-size: 0.875rem;
  font-weight: 600;
  line-height: 1.125rem;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-transform: uppercase;
  cursor: pointer;
  border-style: solid;
  border-width: 1px;
  border-color: transparent;
  border-radius: 4px;
  color: #fff;
  background-color: transparent;
  border-color: #0cd664;
  width: 100%;
}

.emotion-3:disabled,
.emotion-3[disabled] {
  cursor: not-allowed;
  background-color: #414f57;
  color: #677278;
}

.emotion-3.is-active:not([disabled]),
.emotion-3.has-focus:not([disabled]),
.emotion-3:hover:not([disabled]),
.emotion-3:active:not([disabled]),
.emotion-3:focus:not([disabled]) {
  color: #00e865;
  background-color: transparent;
  border-color: #00e865;
}

.emotion-3.has-focus,
.emotion-3:focus {
  outline: none;
  box-shadow: 0 0 16px #0cd664;
}

.emotion-3:disabled,
.emotion-3[disabled] {
  cursor: not-allowed;
  background-color: transparent;
  border-color: #414f57;
  color: #414f57;
}

<div
  className="emotion-5"
>
  <div
    className="emotion-1"
  >
    <h2
      className="emotion-0"
    >
      Need help?
    </h2>
  </div>
  <div
    className="emotion-1"
  >
    <p>
      If you require any assistance our 24/7 live chat support is here to help
    </p>
  </div>
  <div
    className="emotion-1"
  >
    <button
      className="emotion-3"
    >
      Contact Us
    </button>
  </div>
</div>
`;
